pkg/ast/index.go

```diff
@@ -90,6 +90,31 @@ func (i *Index) FirstNodeByNameBytes(name []byte) (Node, bool) {
 	return node[0], true
 }
 
+func (i *Index) FirstNonExtensionNodeByNameBytes(name []byte) (Node, bool) {
+	hash := xxhash.Sum64(name)
+	nodes, exists := i.nodes[hash]
+	if !exists || len(nodes) == 0 {
+		return Node{}, false
+	}
+
+	for j := range nodes {
+		switch nodes[j].Kind {
+		case NodeKindSchemaExtension,
```

Comment:
does it has a sense to add `node.IsExtensionType` helper?

Reply:
also instead of creating new method we could extend `FirstNodeByNameBytes` with a skipper param

smth like
``` go
FirstNodeByNameBytes(name []byte, skipper ...func(node Node) bool)
```

```go
func SkipExtensions(node Node) (skip bool) {
	switch nodes[j].Kind {
	case NodeKindSchemaExtension,
		// all types here
		NodeKindObjectTypeExtension:
		skip = true
	}
    return
}
```

Reply:
this is cool!